src/Workspaces/Remote/Core/RemoteHostAssetSerialization.cs

```diff
@@ -31,27 +31,28 @@ public static async Task WriteDataAsync(ObjectWriter writer, IRemotableDataServi
             if (checksums.Length == 1)
```

Comment:
i'm curiosu why we need to special case the '1' case.  it seems identical to the many case below.  we write the count, then write the checksums...

Reply:
I think this is just to avoid some dictionary allocations. This code will go away soon.